ideascube/forms.py

```diff
@@ -52,3 +60,37 @@ def save(self, *args, **kwargs):
         user.is_staff = True
         user.save()
         return user
+
+
+class UserImportForm(forms.Form):
+    source = forms.FileField(required=True)
+
+    def save(self):
+        source = self.cleaned_data['source']
+        users = []
+        errors = []
+        rows = csv.DictReader(source.read().decode().splitlines())
+        for idx, row in enumerate(rows):
+            serial = row.get('serial')
+            if not serial:
+                continue
+            data = {}
+            for field in User._meta.fields:
+                try:
+                    data[field.name] = row[field.name]
+                except KeyError:
+                    continue
+                if isinstance(field, models.CharField):
+                    data[field.name] = to_unicode(data[field.name])
+
+            instance = User.objects.filter(serial=serial).first()
```

Comment:
That's a weird construction. `serial` has a unicity constraint, so `.filter` doesn't feel like the right thing to use (over `.get`)

Are you doing things this way just to take advantage of the fact that `.filter(...).first()` returns `None` when the filter doesn't match anything?

To me that's relying on a side effect, and it makes the code less obvious than, say:

```python
try:
    instance = User.objects.get(serial=serial)

except User.DoesNotExist:
    instance = None
```

The latter makes it explicit that you want to allow `None` as a valid value for `instance`, and in which case that happens.